src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -455,12 +455,18 @@ static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t
             }
             n++;
         }
+        Value namedCaptures = result->get(state, ObjectPropertyName(state, state.context()->staticStrings().groups)).value(state, result);
         if (functionalReplace) {
             replacerArgs[nCaptures + 1] = Value((size_t)position);
             replacerArgs[nCaptures + 2] = Value(str);
-            replacement = Object::call(state, replaceValue, Value(), nCaptures + 3, replacerArgs).toString(state);
+            size_t argSize = nCaptures + 3;
+            if (!namedCaptures.isUndefined()) {
+                replacerArgs[nCaptures + 3] = namedCaptures;
```

Comment:
`replacerArgs` array is allocated above by statement below
```js
replacerArgs = ALLOCA(sizeof(Value) * (nCaptures + 3), Value, state);
```
which has a fixed length (nCaptures + 3).
But accessing `replacerArgs[nCaptures + 3]` at this point would exceed the array boundary.
So we have to allocate the array length differently. Please check it again.